File: lua/abstract/plugins/extra/Abstract-cs.lua
-- ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ --
-- ───────────────────────────────────────────────── --
--    Plugin:    Abstract-cs
--    Github:    github.com/Abstract-IDE/Abstract-cs
-- ───────────────────────────────────────────────── --
-- ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ --





-- ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ --
-- ━━━━━━━━━━━━━━━━━━━❰ configs ❱━━━━━━━━━━━━━━━━━━━ --
-- ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ --


return {           -- colorscheme for (neo)vim written in lua specially made for Abstract
	"Abstract-IDE/Abstract-cs",
	lazy = false,  -- make sure we load this during startup if it is your main colorscheme
	priority = 1000, -- make sure to load this before all the other start plugins
	config = function()
		-- apply colorscheme without throwing any errors
		local apply_cs, _ = pcall(vim.cmd, 'colorscheme abscs')
		if not apply_cs then return end

		-- theme name
		vim.g.abscs_theme_name = 'aqua'
	end,
}

File: code/src/neovim/editor/src/plugins/configs/abstract_cs.rs
/*
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
─────────────────────────────────────────────────
Plugin: Abstract-cs
Source: https://github.com/Abstract-IDE/Abstract-cs

Colorscheme for (neo)vim written in lua,
specially made for roshnivim with Tree-sitter support.
─────────────────────────────────────────────────
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
*/

use crate::lua_spec;

pub struct Plugin;

impl Plugin {
    pub fn spec() -> crate::plugins::spec::SpecInfo {
        let config = Self::config();
        lua_spec!(format!(
            // language=lua
            r#"{{
                "Abstract-IDE/Abstract-cs",
                branch = "rewrite-2",
                lazy = false,
                priority = 1000,
                config = {config},
            }}"#
        )
        .leak())
    }
}

impl Plugin {
    pub fn config() -> &'static str {
        // language=lua
        r#"function()
            require("abstract_cs").setup()
        end"#
    }
}
